### PR TITLE
Add event delay metric on both producers

### PR DESCRIPTION
**Pull request by dmathieu**
This adds a new header on each producer record, `timestamp`, which stores the time when the record was produced.
It also emits a new metric, `consumer.messages.delay`, which is the delay in milliseconds it took to consume a record.

### CODE REVIEW

File: pubsublite/internal/telemetry/consumer.go [hunk 1]
```diff
@@ -94,7 +112,17 @@ func Consumer(
 		metrics.fetched.Add(context.Background(), 1, metric.WithAttributes(
 			attrs...,
 		))
+
 		if msg.Attributes != nil {
+			if msg.Attributes[apmqueue.EventTimeKey] != "" {
+				if since, err := time.Parse(time.RFC3339, msg.Attributes[apmqueue.EventTimeKey]); err == nil {
+					delay := time.Since(since).Milliseconds()
```
Comment (lahsivjar) [inline pubsublite/internal/telemetry/consumer.go:119]: Unit issue, same as what was raised in previous comment.

File: kafka/metrics.go [hunk 2]
```diff
@@ -67,12 +72,22 @@ func newKgoHooks(mp metric.MeterProvider) (*metricHooks, error) {
 	if err != nil {
 		return nil, formatMetricError(msgFetchedKey, err)
 	}
+
+	messageDelayHistogram, err := m.Int64Histogram(msgDelayKey,
+		metric.WithDescription("The delay between producing messages and reading them"),
+		metric.WithUnit(unitCount),
```
[the remark this answers is not in the document]
Comment (dmathieu) [reply kafka/metrics.go]: You're right, I missed it.

File: kafka/metrics.go [hunk 3]
```diff
@@ -105,9 +120,22 @@ func (h *metricHooks) OnProduceRecordUnbuffered(r *kgo.Record, err error) {
...
+	attrs := attributesFromRecord(r)
+
 	h.messageFetched.Add(context.Background(), 1,
-		metric.WithAttributes(attributesFromRecord(r)...),
+		metric.WithAttributes(attrs...),
 	)
+
+	for _, v := range r.Headers {
+		if v.Key == apmqueue.EventTimeKey {
+			if since, err := time.Parse(time.RFC3339, string(v.Value)); err == nil {
```
Comment (lahsivjar) [inline kafka/metrics.go:131]: Should we log the error, it might help us down the line if something bad happens.

**Review by lahsivjar — APPROVED** [review]
LGTM! A minor comment on logging error.

File: kafka/metrics.go (hunk 3 above)
Comment (dmathieu) [reply kafka/metrics.go]: Good call.
We don't have a logger within any of the telemetry packages. So I've recorded a span error.

**Review by carsonip — APPROVED** [review]

**Review by marclop — COMMENTED** [review]
Is there a reason we're using an additional header to propagate when the record was produced to the queue? Both Kafka and PubSubLite have built-in fields for that purpose. We could be using `record.Timestamp` for Kafka, and `message.PublishTime` for PubSub Lite.

**Comment by dmathieu** [comment]
You're right. I missed that while digging through the docs :(
https://github.com/elastic/apm-queue/pull/222

**Pull request merged**